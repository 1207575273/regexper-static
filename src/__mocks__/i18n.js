import React from 'react';
import i18n from 'i18next';
import { I18nextProvider } from 'react-i18next';

const translate = txt => `translate(${ txt })`;

i18n.init({
  fallbackLng: 'en',
  fallbackNS: 'missing',
  debug: false,
  resources: {}
});

const I18nWrapper = ({ children }) => ( // eslint-disable-line react/prop-types
  <I18nextProvider i18n={ i18n }>
    { React.cloneElement(React.Children.only(children), { t: translate }) }
  </I18nextProvider>
);

export { translate, i18n, I18nWrapper };
