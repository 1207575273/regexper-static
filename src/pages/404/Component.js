import React from 'react';
import PropTypes from 'prop-types';
import { translate, Trans } from 'react-i18next';

import Message from 'components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from 'components/Header';
import Footer from 'components/Footer';

const Component = ({ t }) => (
  <React.Fragment>
    <Header/>
    <Message type="error" icon={ AlertIcon } heading={ t('404 Page Not Found') }>
      <p><Trans>The page you have requested could not be found</Trans></p>
    </Message>
    <Footer/>
  </React.Fragment>
);

Component.propTypes = {
  t: PropTypes.func
};

export default translate()(Component);
