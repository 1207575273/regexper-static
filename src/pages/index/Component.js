import React from 'react';

import Message from 'components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from 'components/Header';
import Footer from 'components/Footer';
import App from 'components/App';

const Component = () => (
  <React.Fragment>
    <Header/>
    <noscript>
      <Message type="error" icon={ AlertIcon } heading="JavaScript Required">
        <p>You need to enable JavaScript to use Regexper.</p>
        <p>If you have concerns regarding the use of tracking code on Regexper, please see the <a href="/privacy.html">Privacy Policy</a>.</p>
      </Message>
    </noscript>
    <App/>
    <Footer/>
  </React.Fragment>
);

export default Component;
