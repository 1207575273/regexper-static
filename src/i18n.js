import i18n from 'i18next';
import { reactI18nextModule } from 'react-i18next';

import locales from './locales';

i18n
  .use(reactI18nextModule)
  .init({
    fallbackLng: 'en',
    debug: (process.env.NODE_ENV !== 'production'),
    resources: locales
  });

export default i18n;
