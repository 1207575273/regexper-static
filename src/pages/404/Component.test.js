import React from 'react';
import { shallow } from 'enzyme';

import { translate } from 'i18n';
import { Component } from './Component';

describe('404 page component', () => {
  test('rendering', () => {
    const component = shallow(
      <Component t={ translate }/>
    );
    expect(component).toMatchSnapshot();
  });
});
