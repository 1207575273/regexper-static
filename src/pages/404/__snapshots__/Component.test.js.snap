// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page component rendering 1`] = `
<React.Fragment>
  <Translate(Header) />
  <Message
    heading="translate(404 Page Not Found)"
    type="error"
  >
    <p>
      <Trans>
        The page you have requested could not be found
      </Trans>
    </p>
  </Message>
  <Translate(Footer) />
</React.Fragment>
`;
